index.html

```diff
@@ -1,725 +1,1449 @@
+
 </p>
 
+
+
 <pre class="language-javascript"><code>
+
 const Frank = xtag.create('x-frankenstein', class extends XTagElement {
+
   'foo::template(true)' (){
+
     return `&lt;h2&gt;My name is Frankenstein&lt;/h2&gt;
+
             &lt;span&gt;I work for a mad scientist&lt;/span&gt;`
+
   }
+
   'bar::template' (){
+
     return `&lt;h2&gt;My friends call me Frank&lt;/h2&gt;
+
             &lt;span&gt;I work for a mad scientist&lt;/span&gt;`
+
   }
+
 });
 
+
+
 const FrankNode = new Frank();
+
 // Because the foo template indicated it was to be automatically rendered, its content is already present in your element instance.
+
 FrankNode.render('bar');
+
 // Calling the render function with the bar template name swaps out the current foo template content with the bar content.
+
 </code></pre>
 
-<hr/>
 
-  <h3>Say Hello to Custom Attribute Accessors</h3>
 
-    <p>
-      Accessors are custom attributes linking to a corresponding pair of setters/getters, and X-Tag has a built-in extension that makes adding accessors to 
-      your custom element definition stupid simple. To use this extension, add an <code>::attr</code> modifier to one of your class definition properties. 
-      This tells X-Tag to wire up a getter/setter to an HTML attribute of the same name. When attributes are linked to a getter/setter their gets, sets, and 
-      state changes will be synced without having to write any additional code.
-    </p>
+<h4>The Elements Render</h4>
+
+<p>
+
+	The XTagElement's <code>render()</code> method is the work horse of the template extension. It get's attached to every X-Tag element that you create and can be reused for your own extensions and pseudos. Below is an example of using an event to initiate a mode accessor.
+
+</p>
+
 
-    <p>
-      You can also tell the accessor extension that type of accessor you are creating, and it will help by sanitizing values and modifying the gets/sets to 
-      match the behavior of the type you declare. I
 
 <pre class="language-javascript"><code>
-xtag.create('x-foo', class extends XTagElement {
-  set 'maxVolume::attr' (value){
-    // X-Tag automatically maps camel cased getter/setter names to their
-    // dashed attribute equivalents. In this example, the `maxVolume` 
-    // getter/setter pair maps to the `max-volume` attribute.
+
+var _rendered = xtag.create( "x-mode", class extends XTagElement {
+
+  set 'mode::attr'(_mode) {
+
+    this.render(_mode);
+
+    return _mode;
+
   }
-});
 
-xtag.create('x-bar', class extends XTagElement {
-  '::template(true)' (){ return '&lt;input /&gt;' }
-  set 'disabled::attr(boolean)' (value){
-    // The ::attr extended property links node.disabled to gets/sets of 
-    // the disabled="" attribute. Because it is declared as a boolean, 
-    // all values passed into its getter/setter will be a Boolean.
-    this.firstElementChild.disabled = value;
+  get 'mode::attr'(){ 
```

Comment:
So basically what I did here was use it the way your template attribute uses it.  I wasn't sure about this myself, did you want to include any  content on these types of uses?  

In the future I can bring up issues before making a PR.

Reply:
Minus the promise api bit of course.

Reply:
I got that mistaken a bit...the render() api will always write the newest template that you summoned into existence...which I thought it wouldn't.  Maybe a line in the docs should be added just in case others misunderstand that templates get replaced rather than appended onto the targeted node.

I.E. the mode attribute itself was so that you could create different types and than define different modes for those types. I have a bit more for x-tag components concerning types and modes.  

I have to push an update to my dashboard build, when I do that I could create a branch with the build that utilizes the x-tags type and mode extension, if you want. It would be helpful, cause I could use some help concerning the event API being used in the extensions being defined. [i.e. they are not firing when added to an extension object]

Reply:
I did more testing with the extensions and got the src plugin running, I guess something maybe out of place somewhere I overlooked.

Reply:
Duh I just found it.

Reply:
I'm sorry for bombing this thread but this is annoying me...I have bit of code working the way its supposed to but it matters what order I put the extensions in. For instance this works =>
`XTagElement.extensions("mode","typed",...)`

but not

`XTagElement.extensions("typed","mode",...)`

Reply:
Is this a bug or is their a reason I'm not seeing?